api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -251,13 +251,45 @@ protected boolean frontLineWorkerAuthorizedForAccess(FrontLineWorker flw, State
         return whitelistService.numberWhitelistedForState(state, flw.getContactNumber());
     }
 
-    protected boolean serviceDeployedInUserState(Service service, State state) {
-        // If I don't have a state for the FLW let them continue further
-        if (state == null) {
+    /**
+     * Check if the service is deployed for a given circle
+     * @param service service to check
+     * @param circle circle of the caller
+     * @return true if circle is null/unknown, or
+     * true if no states in circle or
+     * true if one state in circle is authorized, otherwise
+     * false if none of the states in circle are deployed
+     */
+    protected boolean serviceDeployedInCircle(Service service, Circle circle) {
```

Comment:
No real change here. Moved this method from user controller to base and added comments